presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -125,18 +123,22 @@ private ConnectorSplit toIcebergSplit(FileScanTask task)
                 partitionKeys.put(id, null);
             }
             else {
-                String partitionValue;
-                if (type.typeId() == FIXED || type.typeId() == BINARY) {
-                    // this is safe because Iceberg PartitionData directly wraps the byte array
-                    partitionValue = new String(((ByteBuffer) value).array(), UTF_8);
-                }
-                else {
-                    partitionValue = value.toString();
-                }
-                partitionKeys.put(id, partitionValue);
+                partitionKeys.put(id, partitionValueFromType(type, value));
             }
         });
 
         return Collections.unmodifiableMap(partitionKeys);
     }
+
+    private static String partitionValueFromType(Type type, Object value)
```

Comment:
It looks like this didn't require any changes, so we can leave it as-is

Reply:
Change reverted.